lib/model.js

```diff
@@ -478,7 +478,7 @@ assign(derived, {
           authMechanism: this.driverAuthMechanism
         });
       } else if (this.authStrategy === 'X509') {
-        req.auth = this.x509Username;
+        req.auth = AUTH_TOKEN;
```

Comment:
Here we use a placeholder to build URI without special characters at all, and later we replace it with encoded auth.